### PR TITLE
[CI] Refactor pipelines

### PR DESCRIPTION
**Pull request by CedricGuillemet**
- Refactor azure-pipelines.yml to get rid of copy/pasted jobs
- Install CMake MacOS without Homebrew direct link as Bintray is stopping the service

### CODE REVIEW

File: .github/jobs/macos_cmake.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+parameters:
+  vmImage: ''
+
+steps:
+  - script: |
+      brew uninstall cmake
```
Comment (Drigax) [inline .github/jobs/macos_cmake.yml:6]: Looks like this should workaround https://github.com/BabylonJS/BabylonNative/issues/683, thanks!

Comment (CedricGuillemet) [reply .github/jobs/macos_cmake.yml]: Same for ninja which is also on homebrew

File: .github/jobs/macos_cmake.yml [hunk 2]
```diff
@@ -0,0 +1,15 @@
+parameters:
+  vmImage: ''
+
+steps:
+  - script: |
+      wget https://github.com/Kitware/CMake/releases/download/v$(CMAKE_VERSION)/cmake-$(CMAKE_VERSION)-macos-universal.tar.gz
```
Comment (Drigax) [inline .github/jobs/macos_cmake.yml:6]: I noticed that my dev machine does not have wget, adding some snippets so all our bases are covered for installation and reinstallation, unless we're already doing this in a previous step.
```suggestion
        brew install wget
        wget https://github.com/Kitware/CMake/releases/download/v$(CMAKE_VERSION)/cmake-$(CMAKE_VERSION)-macos-universal.tar.gz
```

File: .github/jobs/macos_cmake.yml [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+      sudo mkdir -p /opt/cmake
```
Comment (Drigax) [inline .github/jobs/macos_cmake.yml:7]: do we want to remove /opt/cmake if it already exists?
```suggestion
      sudo rm -rf /opt/cmake
      sudo mkdir -p /opt/cmake
```

File: .github/jobs/macos_cmake.yml (hunk 2 above)
Comment (CedricGuillemet) [reply .github/jobs/macos_cmake.yml:6]: wget is available by default on all VMs. Do it locally only.

File: .github/jobs/macos_cmake.yml (hunk 3 above)
Comment (CedricGuillemet) [reply .github/jobs/macos_cmake.yml:7]: /opt should be empty by default in the VMs

File: .github/jobs/macos_cmake.yml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+      sudo tar xvzf cmake-$(CMAKE_VERSION)-macos-universal.tar.gz -C /opt/cmake
+      sudo ln -s /opt/cmake/cmake-$(CMAKE_VERSION)-macos-universal/CMake.app/contents/bin/cmake /usr/local/bin/cmake
```
[the remark this answers is not in the document]
Comment (CedricGuillemet) [reply .github/jobs/macos_cmake.yml:10]: yes, done.

File: .github/jobs/macos_cmake.yml (hunk 2 above)
Comment (Drigax) [reply .github/jobs/macos_cmake.yml:6]: cool, not needed.

File: .github/jobs/macos_cmake.yml (hunk 3 above)
Comment (Drigax) [reply .github/jobs/macos_cmake.yml:7]: Ok, I wasn't sure what the lifecycle for a VM was, if it was recreated from an image for each pipeline run, or only periodically

**Review by bghgary — APPROVED** [review]

**Pull request merged**